### PR TITLE
Add Permission Class on Enrolment API

### PR DESCRIPTION
**Pull request by burhandodhy**
**Description**
This PR adds the permission class that allows the user to only access their site

**Jira Ticket** 
https://edlyio.atlassian.net/browse/EDLY-4220

### CODE REVIEW

File: openedx/features/edly/permissions.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+"""
+Custom permissions for edly.
+"""
+from rest_framework.permissions import BasePermission
+
+
+class CanAccessEdxAPI(BasePermission):
+    """
+    Checks if a user ca access Edx API.
```
Comment (madnihamza1841) [inline openedx/features/edly/permissions.py:9]: Typing mistake.  ca --> can

**Review by madnihamza1841 — APPROVED** [review]
LGTM. 
Just a typing mistake found. 

"Checks if a user ca access Edx API." --> "Checks if a user can access Edx API."

File: openedx/features/edly/permissions.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    """
+
+    def has_permission(self, request, view):
+        return request.user.is_staff or request.user.edly_profile.edly_sub_organizations.filter(
+            lms_site=request.site
+        ).exists()
```
Comment (taimoor-ahmed-1) [inline openedx/features/edly/permissions.py:15]: for second clause can we write it like this?
```request.site.edly_sub_org_for_lms.slug in request.user.edly_profile.get_linked_edly_sub_organizations```

File: openedx/core/djangoapps/enrollments/views.py [hunk 3]
```diff
@@ -36,6 +36,7 @@
 from openedx.core.lib.exceptions import CourseNotFoundError
 from openedx.core.lib.log_utils import audit_log
 from openedx.features.edly.utils import is_course_org_same_as_site_org
+from openedx.features.edly.permissions import CanAccessEdxAPI
```
Comment (taimoor-ahmed-1) [inline openedx/core/djangoapps/enrollments/views.py:39]: import should be above `edly.utils`

**Review by taimoor-ahmed-1 — CHANGES_REQUESTED** [review]

File: openedx/core/djangoapps/enrollments/views.py (hunk 3 above)
Comment (burhandodhy) [reply openedx/core/djangoapps/enrollments/views.py:38]: Done

File: openedx/features/edly/permissions.py (hunk 2 above)
Comment (burhandodhy) [reply openedx/features/edly/permissions.py]: Done

**Review by taimoor-ahmed-1 — APPROVED** [review]
LGTM

File: openedx/features/edly/permissions.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+"""
+from rest_framework.permissions import BasePermission
+
+
+class CanAccessEdxAPI(BasePermission):
+    """
+    Checks if a user can access Edx API.
+    """
+
+    def has_permission(self, request, view):
```
Comment (ziafazal) [inline openedx/features/edly/permissions.py:12]: @burhandodhy this permission is not enough we should make sure only previliged users or only those users who belong to a particular ground should be able to enroll other users then themselves. With current permission any user can enroll anyother user as long as both belong to same site.

Comment (burhandodhy) [reply openedx/features/edly/permissions.py:12]: @ziafazal this permission class only restrict that user of one site can't request on the other class. The class is written in a generic way so we could the same class for the other API in future.

As to restrict the enrollment @taimoor-ahmed-1 has created a separate PR https://github.com/edly-io/edx-platform/pull/300/files

**Pull request merged**